### PR TITLE
fixup! ASoC: SOF: ipc4: fixup dailink based on copier format

### PR DESCRIPTION
**Pull request by ujfalusi**
There is really no need to do fixup in case of ChainDMA, just return.

The 'ASoC: SOF: ipc4: fixup dailink based on copier format` causes 16bit audio to be broken.

### CODE REVIEW

**Review by jsarha — DISMISSED** [review]
Makes sense and fixes the problem with 16-bit HDMI audio playback.

File: sound/soc/sof/ipc4-pcm.c [hunk 1]
```diff
@@ -571,12 +571,12 @@ static int sof_ipc4_pcm_dai_link_fixup(struct snd_soc_pcm_runtime *rtd,
 	}
 
 	/* Chain DMA does not use copiers, so no fixup needed */
-	if (!use_chain_dma) {
-		int ret = sof_ipc4_pcm_dai_link_fixup_rate(sdev, params, ipc4_copier);
+	if (use_chain_dma)
+		return 0;
```
Comment (aiChaoSONG) [inline sound/soc/sof/ipc4-pcm.c]: How about eliminate the use of use_chain_dma variable and bring
```
if (pipeline->use_chain_dma)
   return 0;
```
to the beginning of the for_each_pcm_streams loop? so the code that make no sense under `pipeline->use_chain_dma == true` is not executed.

**Review by bardliao — DISMISSED** [review]

File: sound/soc/sof/ipc4-pcm.c (hunk 1 above)
Comment (ujfalusi) [reply sound/soc/sof/ipc4-pcm.c]: I thought about that, we can do it, but I wanted to have the smallest change possible which can be squashed into the offending commit.
A bigger change should be a separate one and it should be moved earlier as a simplification for the ChainDMA handling. For upstream, that is.

**Comment by ujfalusi** [comment]
Changes since v1:
- drop out from the loop and return if ChainDMA is used (@aiChaoSONG, it does not look too bad after all)

**Review by bardliao — APPROVED** [review]

**Review by plbossart — APPROVED** [review]

**Review by aiChaoSONG — APPROVED** [review]

**Comment by plbossart** [comment]
test results https://sof-ci.01.org/linuxpr/PR4520/build6098/devicetest/index.html and https://sof-ci.01.org/linuxpr/PR4520/build6099/devicetest/index.html are not visible, huh?

**Comment by plbossart** [comment]
SOFCI TEST

**Comment by aiChaoSONG** [comment]
SOFCI TEST

**Pull request merged**